src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/DeconstructExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.UpgradeAssistant
+{
+    internal static class DeconstructExtensions
```

Comment:
Is this used anywhere? I don't see references to it.

Reply:
It's used to extend the deconstruction so we can foreach over keyvaluepairs in a few places and deconstruct to tuples